### ISSUE TITLE
Checkstyle settings does not allow to @SuppressWarnings

### ISSUE THREAD

**Issue by Jan-PeterRichterCG** [devonfw/ide-settings#3]
Note: this enhancement required #1 to be fixed!

As a developer, I want supress some CheckStyle warnings if they cannot be fixed easily or are false-positives so that the list of warnings always shows relevant entries only.

As documented on stackoverflow here https://stackoverflow.com/questions/27688426/ignoring-of-checkstyle-warnings-with-annotation-suppresswarnings , the checkstyle.xml must contain the modules SuppressWarningsHolder and SuppressWarningsFilter:
```
<module name="Checker">
  <--! ... -->
  <module name="TreeWalker">
    <--! ... -->
    <module name="SuppressWarningsHolder"/>
  </module>
  <module name="SuppressWarningsFilter"/>
</module>
```
(I have played around with these modules: yes both have to be present and eclipse has to be started anew and the project has to be build again so that changes take effect for the complete project)

When these lines are added to checkstyle.xml a developer can add an annotation like
``` @SuppressWarnings({ "checkstyle:MagicNumber", "checkstyle:CyclomaticComplexity" }) ```
before a class definition and the checks are disabled for that class.

**Comment by hohwille** [comment]
I did not take notice of issues in this repository. I will try to resolve them and then close down the "issue" tab here so issues are created in the `ide` repository itself where we can assign labels and milestones, etc.

**Comment by hohwille** [comment]
For your feature:
Please note that we have already configured suppression comment filter:
https://github.com/devonfw/ide-settings/blob/5464374af1b028af732e2ce807d13dbe37d8ed30/checkstyle/checkstyle.xml#L131

So you can simply add inline comments like:
```
// CHECKSTYLE:OFF (explain here why you are doing this)
...
// CHECKSTYLE:ON
```

Comparing comment vs. annotation for suppression:
* +comment: using the annotation is causing issues with other toolings that raise warnings if unknown supressions are used for `java.lang.SuppressWarnings` and using proprietary annotations is even worse as you have to add a dependency to all your code just for the supression including APIs, etc. consumed by third-parties
* +comment: with an annotation you can not as easily and flexbily define the range where you want to suppress
* +annotation: with an annotation you can easily suppress for the range of a type or method and you can not forget to also add a corresponding `// CHECKSTYLE:OFF` what would lead to disable it for the entire rest of the file.

So in the end both have their pros and cons. For simplicity I will just add the config as you proposed.
To speed up things like this people can at any time create pull-requests with the proposed changes rather than adding an issue with discription in which file to change what. See our contribution guide for details:
https://github.com/devonfw/.github/blob/master/CONTRIBUTING.asciidoc#contributing
Ho

**Comment by hohwille** [comment]
I can not assign milestone from here but this will be released with `2020.08.001`.
